### DISCUSSION TITLE
Can't upload images to storage programatically

### DISCUSSION THREAD

**Discussion by Dhaiwat10** [supabase/supabase#2682] [Questions] [answered]
When I try to upload images to Storage using the Javascript library, this is the response that I get:

```
{"statusCode":"42501","error":"","message":"new row violates row-level security policy for table \"objects\""}
```

My request:
<img width="489" alt="Screenshot 2021-08-03 at 4 53 50 AM" src="https://user-images.githubusercontent.com/39617427/127935538-c9fe00bf-5b27-47e2-a8db-0dbefb3674d8.png">

My code:
```javascript
// file is a File, listingId a string, idx a number
export async function uploadImage(file, listingId, idx) {
  const fileExt = file.name.split('.').pop();
  const fileName = `${idx}.${fileExt}`;
  const filePath = `${listingId}/${fileName}`;

  let { error: uploadError } = await supabase.storage
    .from('public')
    .upload(filePath, file);

  if (uploadError) {
    return false;
  }

  return true;
}
```

I'd appreciate some help. :)

**Comment by Dhaiwat10** [comment]
I am being able to upload images from the UI, though. 🤔

**Comment by inian** [comment, marked as answer]
Hi @Dhaiwat10, you need to have the correct storage policies set to configure who can upload objects to your buckets. If you want to bypass all permissions, you can use the service key, but anyone who also has access to the service key can also do the same. So this is fine if you are using it in your backend, but not in your frontend application.

**Reply by Dhaiwat10** [reply]
I see. Thanks for the explanation.
